src/handle_check_address.c

```diff
@@ -18,76 +19,47 @@ void handle_check_address(check_address_parameters_t* params, chain_config_t* ch
     const uint8_t* bip32_path_ptr = params->address_parameters;
     uint8_t bip32PathLength = *(bip32_path_ptr++);
     cx_sha3_t local_sha3;
-
-    // Common memory is used for locals that are not used concurrently
-    union group1 {
-        uint32_t bip32Path[MAX_BIP32_PATH];
-        cx_ecfp_private_key_t privateKey;
-        char address[51];
-    } locals_union1;
-    union group2 {
-        uint8_t privateKeyData[64];
-        cx_ecfp_public_key_t publicKey;
-    } locals_union2;
+    uint32_t bip32Path[MAX_BIP32_PATH];
+    char address[51];
+    cx_ecfp_256_private_key_t priv = {0};
+    cx_ecfp_256_public_key_t pub = {0};
+    cx_err_t error = CX_INTERNAL_ERROR;
 
     if ((bip32PathLength < 0x01) || (bip32PathLength > MAX_BIP32_PATH) ||
         (bip32PathLength * 4 != params->address_parameters_length - 1)) {
         PRINTF("Invalid path\n");
         return;
     }
     for (uint8_t i = 0; i < bip32PathLength; i++) {
-        locals_union1.bip32Path[i] = U4BE(bip32_path_ptr, 0);
+        bip32Path[i] = U4BE(bip32_path_ptr, 0);
         bip32_path_ptr += 4;
     }
-    if (os_derive_bip32_no_throw(CX_CURVE_256K1,
-                                 locals_union1.bip32Path,
-                                 bip32PathLength,
-                                 locals_union2.privateKeyData,
-                                 NULL) != CX_OK) {
-        ZERO(locals_union1);
-        ZERO(locals_union2);
-        return;
-    }
+    CX_CHECK(bip32_derive_with_seed_init_privkey_256(HDW_NORMAL,
```

Comment:
Any reason not to sue bip32_derive_get_pubkey_256?

Reply:
Done!